Fixes some problems I've found using ffi.el

zevlg opened the pull request:
[fix] define-ffi-struct: make it work for definitions like

    (define-ffi-struct PointInfo
      (x :type :double)
      (y :type :double))

[fix] with-ffi-string/with-ffi-temporary: execute body under `progn',
      so multiexpression body won't screw things up

zevlg committed:
[fix] define-ffi-struct: make it work for definitions like

    (define-ffi-struct PointInfo
      (x :type :double)
      (y :type :double))

[fix] with-ffi-string/with-ffi-temporary: execute body under `progn',
      so multiexpression body won't screw things up

tromey commented on ffi.el:
```diff
@@ -98,8 +98,8 @@ SLOT-NAME is a symbol and TYPE is an FFI type descriptor."
-	 (apply #'ffi--define-struct (make-list (eval length)
-						(symbol-name type)))))
+	 (apply #'ffi--define-struct
+                (make-list (eval length) type))))
```
I don't remember what's going on here - when does the symbol-name matter or not?

zevlg replied:
It possible should be `symbol-value` instead of `symbol-name`, making user defined types are also valid for array definition

I'll check later on

tromey commented:
Thanks!  I think this looks pretty good, but would you mind adding a test case?

zevlg committed:
[fix] use `symbol-value' to get type in array definition

[add] tests added for few cases

zevlg commented:
I figured out about `symbol-name`.  It definitely was just a typo, `symbol-value` should be used instead.

fixed, and I also added test cases https://github.com/tromey/emacs-ffi/pull/13/commits/db05bb4cdaceeebd01bf5f47c5d13755e87c4099

tromey commented:
Thanks!
